lib/launcher.rb

```diff
@@ -149,32 +177,85 @@ def write_pid
       retry
     end
 
-    def webrick_server(app, addresses, port)
+    def add_puma_server(app, addresses, port, conn_type)
+      # IMPORTANT:
+      # The following code takes only a single host.
+      # The current reason for it, is that "run" is blocking, and in order to
+      # add support for more hosts, additional threads requires to be created
+      address = addresses.first
+      address = '0.0.0.0' if address == '*'
+      if conn_type == :ssl
+        host = "ssl://#{address}/"
```

Comment:
I'd drop this line and just write `host = "ssl://#{address}/?#{query_list.join('&')}"` below

Reply:
Sure.